light-client/src/components/scheduler.rs

```diff
@@ -122,5 +123,7 @@ pub fn valid_schedule(
 #[pre(low <= high)]
 #[post(low <= ret && ret <= high)]
 fn midpoint(low: Height, high: Height) -> Height {
-    (low.value() + (high.value() + 1 - low.value()) / 2).into()
+    (low.value() + (high.value() + 1 - low.value()) / 2)
+        .try_into()
+        .unwrap() // Will panic if midpoint is higher than i64::MAX
```

Comment:
Good idea, although in this case it's more of a theoretical possibility than a real one.
You can get the biggest number with `high.value()==i64::MAX` and midpoint is going to be `i64::MAX/2` in that case. So this is an unwrap case that can never fail, unless the developer changes the types without concern to the note.

Maybe it's better to remove it, since it's theoretical only.